packages/cli/src/oclif/commands/scaffold.js

```diff
@@ -0,0 +1,213 @@
+const _ = require('lodash');
+const path = require('path');
+const colors = require('colors/safe');
+
+const { flags } = require('@oclif/command');
+
+const BaseCommand = require('../ZapierBaseCommand');
+const { buildFlags } = require('../buildFlags');
+const {
+  ensureDir,
+  fileExistsSync,
+  readFile,
+  writeFile
+} = require('../../utils/files');
+const { camelCase, snakeCase } = require('../../utils/misc');
+const { splitFileFromPath } = require('../../utils/string');
+
+// what is the `resources: {}` app definition point?
+const typeMap = {
+  resource: 'resources',
+  trigger: 'triggers',
+  search: 'searches',
+  create: 'creates'
+};
+
+function createTemplateContext(type, name) {
+  const contextKey = snakeCase(name);
+
+  // where will we create/required the new file?
+  const destMap = {
+    resource: `resources/${contextKey}`,
+    trigger: `triggers/${contextKey}`,
+    search: `searches/${contextKey}`,
+    create: `creates/${contextKey}`
+  };
+
+  return {
+    CAMEL: camelCase(name),
+    KEY: contextKey,
+    NOUN: _.capitalize(name),
+    LOWER_NOUN: name.toLowerCase(),
+    INPUT_FIELDS: '',
+    TYPE: type,
+    TYPE_PLURAL: type === 'search' ? `${type}es` : `${type}s`,
+    // resources need an extra line for tests to "just run"
+    MAYBE_RESOURCE: type === 'resource' ? 'list.' : '',
+    defaultDest: destMap[type]
+  };
+}
+
+function getTemplatePath(type) {
+  return path.join(__dirname, '../../..', `scaffold/${type}.template.js`);
+}
```

Comment:
I liked the feel of pulling these functions out of the class. They're not exactly utils worthy of their own file, but could be class methods if we really wanted.

Reply:
yeah, either way! I think it's reasonable to bundle them all together. Typically we could test them, though these are simple enough we probably won't bother.